src/temgymbasic/components.py

```diff
@@ -146,6 +155,114 @@ def gui_wrapper():
         return SampleGUI
 
 
+class PotentialSample(Sample):
+    def __init__(
+        self,
+        z: float,
+        potential: NDArray,
+        pixel_size: float,
+        shape: Tuple[int, int],
+        rotation: Radians = 0.,
+        flip_y: bool = False,
```

Comment:
`flip_y` / `center` / `rotation` / `pixel_size` should be removed for now.

It might actually be worth having a light `Potential` class inspired by abTEM which implements the interpolation etc, that way the `Component` remains focused on the interface to model. Probably a future PR though.

Reply:
This is added now in commit 
0f3e225

Reply:
at least the removal of those inputs. The potential class being added in would give an extent to this model, which could then be used for plotting in matplotlib, so that's one reason to use it. I do not know how to typehint that potential, Ex and Ey should be RegularGridInterpolant class from scipy, but that would be important I think.

Reply:
At the moment this just does the absolute basics that potential needs, with a sample centred on 0,0. I think it's time for me to write a test for this, that the deflections are actually correct and match the force that the equation  Fx = qEx, Fy = qEy, says should happen.